src/utils/tests/getElementByDataTestIdDeep.puppeteer.ts

```diff
@@ -0,0 +1,29 @@
+import { createDataTestIdSelector, walkShadowDom } from './shadowDomTraversal';
+
+/**
+ * Recursively searches the document and all shadow roots for the first element
+ * matching the given data-testid.
+ *
+ * @param {import('puppeteer').Page} page - The Puppeteer Page instance.
+ * @param {string} dataTestId - The value of the data-testid attribute to search for.
+ * @returns {Promise<import('puppeteer').ElementHandle|null>} - Handle to the found element, or null.
+ */
+export async function getElementByDataTestIdDeepPuppeteer(
+  page: any,
```

Comment:
Same comments as above, the three points mentioned on the other helper.

Reply:
To avoid name collision on exports and imports